cmd/maestro/server/grpc_server.go

```diff
@@ -0,0 +1,172 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/binding"
+	cloudeventstypes "github.com/cloudevents/sdk-go/v2/types"
+	"github.com/golang/glog"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/protobuf/types/known/emptypb"
+	pbv1 "open-cluster-management.io/api/cloudevents/generic/options/grpc/protobuf/v1"
+	grpcprotocol "open-cluster-management.io/api/cloudevents/generic/options/grpc/protocol"
+	"open-cluster-management.io/api/cloudevents/generic/types"
+	"open-cluster-management.io/api/cloudevents/work/payload"
+
+	"github.com/openshift-online/maestro/pkg/api"
+	"github.com/openshift-online/maestro/pkg/config"
+	"github.com/openshift-online/maestro/pkg/services"
+)
+
+// GRPCServer includes a gRPC server and a resource service
+type GRPCServer struct {
+	pbv1.UnimplementedCloudEventServiceServer
+	grpcServer      *grpc.Server
+	resourceService services.ResourceService
+	bindAddress     string
+}
+
+// NewGRPCServer creates a new GRPCServer
+func NewGRPCServer(resourceService services.ResourceService, config config.GRPCServerConfig) *GRPCServer {
+	grpcServerOptions := make([]grpc.ServerOption, 0)
+	grpcServerOptions = append(grpcServerOptions, grpc.MaxRecvMsgSize(config.MaxReceiveMessageSize))
+	grpcServerOptions = append(grpcServerOptions, grpc.MaxSendMsgSize(config.MaxSendMessageSize))
+	grpcServerOptions = append(grpcServerOptions, grpc.MaxConcurrentStreams(config.MaxConcurrentStreams))
+	grpcServerOptions = append(grpcServerOptions, grpc.ConnectionTimeout(config.ConnectionTimeout))
+	grpcServerOptions = append(grpcServerOptions, grpc.WriteBufferSize(config.WriteBufferSize))
+	grpcServerOptions = append(grpcServerOptions, grpc.ReadBufferSize(config.ReadBufferSize))
```

Comment:
Consider adding MAX_CONNECTION_AGE to optimize load balancing for prolonged client-server connections.